spec/models/user_spec.rb

```diff
@@ -6,5 +6,32 @@
   describe 'validations' do
     it { should validate_presence_of(:email) }
     it { should validate_presence_of(:password) }
+    it { is_expected.to have_db_column(:email).of_type(:string).with_options(default: '', null: false) }
+    it { is_expected.to have_db_column(:phone_number).of_type(:string) }
+    it { is_expected.to have_db_column(:role).of_type(:integer).with_options(default: :user) }
+    it { is_expected.to have_db_column(:encrypted_password).of_type(:string).with_options(default: '', null: false) }
+    it { is_expected.to have_db_column(:reset_password_token).of_type(:string) }
+    it { is_expected.to have_db_column(:reset_password_sent_at).of_type(:datetime) }
+    it { is_expected.to have_db_column(:created_at).of_type(:datetime).with_options(precision: 6, null: false) }
+    it { is_expected.to have_db_column(:updated_at).of_type(:datetime).with_options(precision: 6, null: false) }
+    it { is_expected.to have_db_index(:email).unique(true) }
+    it 'creates valid user' do
+      expect(create(:user)).to be_valid
+      expect { create(:user) }.to change(User, :count).by(1)
+    end
+
+    it 'does not creates invalid user' do
```

Comment:
create a context where you will define `invalid_data` like let variable
and` let(:user) { build(:user, invalid_data) }`